File: HackerRank/Data Structure/Linked Lists/Insert a Node at the Tail of a Linked List.cpp
Node* Insert(Node *head,int data)
{
  // Complete this method
    Node*newnode=(Node*)malloc(sizeof(Node));
    newnode->data=data;
    newnode->next=NULL;
    if(head==NULL)
    {
    head=newnode;
        
    }
    else
    {
    Node*temp=head;
    while(temp->next!=NULL)
    {
    temp=temp->next;
    
    }
        temp->next=newnode;
    
    }
    return head;
}

File: HackerRank/Data Structure/Linked Lists/Insert a Node at the Tail of a Linked List.java
static SinglyLinkedListNode insertNodeAtTail(SinglyLinkedListNode head, int data) {
            
            SinglyLinkedListNode temp = head;
            if(temp == null){
                SinglyLinkedListNode new_node = new SinglyLinkedListNode(data);
                head = new_node;
                return head;
            }    
        
            while(temp.next != null) {
                System.out.println(temp.data);
                temp = temp.next;
            }
            
            SinglyLinkedListNode new_node = new SinglyLinkedListNode(data);
            temp.next = new_node;
        
        return head;

    }